addons/mail/static/src/messaging/component/follower_list_menu/follower_list_menu.js

```diff
@@ -0,0 +1,168 @@
+odoo.define('mail.messaging.component.FollowerListMenu', function (require) {
+'use strict';
+
+const components = {
+    Follower: require('mail.messaging.component.Follower'),
+};
+const useStore = require('mail.messaging.component_hook.useStore');
+
+const { Component } = owl;
+const { useRef, useState } = owl.hooks;
+
+class FollowerListMenu extends Component {
+    /**
+     * @override
+     * @param {...any} args
+     */
+    constructor(...args) {
+        super(...args);
+        this.state = useState({
+            hasDropdown: false,
+            hasUnfollowHover: false,
+        });
+        useStore(props => {
+            const thread = this.env.entities.Thread.get(props.threadLocalId);
+            const followers = thread ? thread.followers : [];
+            const isCurrentPartnerFollowing = thread ? thread.isCurrentPartnerFollowing : false;
+            return [
+                followers.length,
```

Comment:
I don't know, I wonder if this is something I did or something Alex did ... With that logic and the logic of the comment above, the useStore could just return the thread, no ?

Reply:
I think it doesn't work for relations due to them being getters or something like that, so we still have to explicitly observe them. `use_store_tests` should be completed (not on this PR) if there are cases that aren't clear, especially for entities, there is no test there at all.

Reply:
Ok then all keys are needed since isCurrentPartnerFollowing is a getter on thread, right ?
Then maybe discuss with @alexkuhn about the length of followers because I cannot answer to you on that ...

Reply:
Well I'm pretty sure it is a bug to just observe length when `localId` of each element is used in the code.

If we have a situation were the length is the same but the elements become different it will not rerender as it should.

Just need to observe `followers` (not length) with `compareDepth`. Change it like that and if it still works then it's good enough. We will probably do other passes on `useStore` at some point anyway.

Reply:
agree